File: vendaval.client/src/app/order/order.ts
import { Product } from "../product/product";
import { BaseModel } from "../shared/common/interfaces/BaseModel";
import { OrderStatus } from "./orderStatus";
import { PaymentMethod } from "./paymentMethod";
import { Address } from "../shared/common/interfaces/address"; 
import { DeliveryType } from "./deliveryType";
export interface Order extends BaseModel{
  costumerId: number;
  products: Product[];
  total: number;
  installments: number;
  installmentsValue: number;
  status: OrderStatus;
  paymentDate: Date | null;
  paymentMethod: PaymentMethod;
  deliveryAddress: Address | null;
  deliveryType: DeliveryType;
  trackingCode: string | null;
  orderNotes: string | null;
}

File: Vendaval.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Vendaval.Domain.Enums;
using Vendaval.Domain.ValueObjects;

namespace Vendaval.Domain.Entities
{
    public class Order : BaseModel
    {
        public int CustomerId { get; set; }
        public required List<Product> Products { get; set; }
        public decimal Total { get; set; }
        public OrderStatus Status { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public int Installments { get; set; }
        public decimal InstallmentValue { get; set; }
        public DateTime? PaymentDate { get; set; }
        public Address? DeliveryAddress { get; set; }
        public DeliveryType DeliveryType { get; set; }
        public string? TrackingCode { get; set; }
        public string? OrderNotes { get; set; }
    }
}
